app.rb

```diff
@@ -0,0 +1,260 @@
+    if person_class == '1'
+      print 'Classroom: '
+      student_classroom = gets.chomp
+      print 'Has parent permission? [Y/N]: '
+      parent_permission = gets.chomp.downcase
+      parent_permission = parent_permission == 'y'
+      new_student = Student.new(result_arr[0], result_arr[1], student_classroom, parent_permission: parent_permission)
+      @people << new_student
+    else
+      print 'Specialization: '
+      teacher_specialization = gets.chomp
+      new_teacher = Teacher.new(result_arr[0], result_arr[1], teacher_specialization)
+      @people << new_teacher
+    end
+    puts 'Person created successfully'
+  end
+
+  def getting_name_age
+    common_arr = []
+    print 'Age: '
+    person_age = gets.chomp.to_i
+    print 'Name: '
+    person_name = gets.chomp
+    common_arr << person_age
+    common_arr << person_name
+  end
+
+  def list_all_people
+    @people.each do |person_item|
+      puts "[#{person_item.class}] Name: #{person_item.name}, ID: #{person_item.id}, Age: #{person_item.age}"
+    end
+  end
+
+  def create_rental
+    puts 'Select a book from the following list by number'
+    @books.each_with_index { |book, index| puts "#{index}) Title: \"#{book.title}\", Author: #{book.author}" }
+    book_index = gets.chomp.to_i
+    puts 'Select a person from the following list by number (not id)'
+    @people.each_with_index do |person_item, index|
+      puts "#{index}) [#{person_item.class}] Name: #{person_item.name}, ID: #{person_item.id}, Age: #{person_item.age}"
+    end
+    person_index = gets.chomp.to_i
+    print 'Date: '
+    rental_date = gets.chomp
+    @rentals << Rental.new(rental_date, @books[book_index], @people[person_index])
+    puts 'Rental created successfully'
+  end
+
+  def list_person_rental
+    print 'ID of person: '
+    person_id = gets.chomp.to_i
+    rentals_result = @rentals.select { |rental_item| rental_item.person.id == person_id }
+    puts 'Rentals:'
+    rentals_result.each do |rental_result|
+      puts "Date: #{rental_result.date}, Book \"#{rental_result.book.title}\" by #{rental_result.book.author}"
+    end
+  end
+end
+
+def present_options
+  puts "\nPlease choose an option by entering a number:"
+  puts '1 - List all books'
+  puts '2 - List all people'
+  puts '3 - Create a person'
+  puts '4 - Create a book'
+  puts '5 - Create a rental'
+  puts '6 - List all rentals for a given person id'
+  puts '7 - Exit'
+  gets.chomp.to_i
+end
+
+def start_method(app, user_option)
+  methods_list = {
+    1 => :list_all_books,
+    2 => :list_all_people,
+    3 => :create_person,
+    4 => :create_book,
+    5 => :create_rental,
+    6 => :list_person_rental,
+    7 => :exit
+  }
+  app.send(methods_list[user_option])
+end
+
+def main
+  app = App.new
+  puts 'Welcome to School Library App!'
+  loop do
+    option = present_options
+    if option == 7
+      puts 'Thank you for using this app!'
+      break
+    end
+    start_method(app, option)
+  end
+end
+
+main
```

Comment:
- [ ] Please refactor this part of the executed code to exist in the `main.rb` file as per project requirements [here](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_ui.md#:~:text=In%20your%20main.rb%20define%20the%20entry%20point%2C%20this%20will%20be%20a%20method%20called%20main). This is for a cleaner and more readable code base. 🙏